R/analysis_page.R

```diff
@@ -470,6 +490,48 @@ AnalysisPageServer <- function(id, data) {
       }
     )
     
+    #Run the model in the server (i.e. not inside an output) and obtain the summary
+    model_summary <- reactive({
+      newData <- data()
+      b <- study_level_outcomes(newData) # get sens, spec for each trial
+      bb <- data.frame(Author=newData$author, Year=newData$year, TP=newData$TP, FN=newData$FN, FP=newData$FP, 
+                       TN=newData$TN, N=b$N, Sens=b$Sens, Spec=b$Spec)
+      bb$Sens <- sprintf('%4.3f', bb$Sens) # restrict number of figures after decimal place for sens
+      bb$Spec <- sprintf('%4.3f', bb$Spec)
+      # add information about percentage weights 
+      N <- length(newData$TP)
+      newData$n1 <- newData$TP+newData$FN
+      newData$n0 <- newData$FP+newData$TN
+      newData$true1 <- newData$TP
+      newData$true0 <- newData$TN 
+      newData$study <- 1:N
+      Y = reshape(newData, direction = "long", varying = list( c("n1" , "n0") , c( "true1","true0" ) ) ,
+                  timevar = "sens" , times = c(1,0) , v.names = c("n","true") ) 
+      Y = Y[order(Y$id),]  
+      Y$spec<- 1-Y$sens
+      MA_Y = glmer( formula = cbind(  true , n - true ) ~ 0 + sens + spec + (0+sens + spec|study),
+                    data = Y , family = binomial , nAGQ=1 , verbose=0 ) 
+      return(summary(MA_Y))
+    })
+    
+    #Check if the model converged
+    converged <- reactive(is.null(model_summary()$optinfo$conv$lme4$messages))
+    
+    #Add converged() reactive to the output list
+    output$converged <- reactive(converged())
+    outputOptions(x = output, name = "converged", suspendWhenHidden = FALSE)
+    
+    #If the model did not converge, display a warning
+    observe({
+      if (!converged()) {
+        shinyalert(
+          title = "WARNING - model failed to converge",
+          text = paste0("Most output will not be displayed. \n\n MetaDTA is not equipped for model diagnostics. You are advised to use other software. \n\nError message: \n", model_summary()$optinfo$conv$lme4$messages),
```

Comment:
Is it worth pointing the user to MetaBayesDTA?

Reply:
Yes, I forgot about that.